app/components/settings/billing-page/payment-history-section.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import * as Sentry from '@sentry/ember';
+import rippleSpinnerImage from '/assets/images/icons/ripple-spinner.svg';
+import type Store from '@ember-data/store';
+import type UserModel from 'codecrafters-frontend/models/user';
+import type ChargeModel from 'codecrafters-frontend/models/charge';
+
+interface Signature {
+  Element: HTMLDivElement;
+  Args: {
+    user: UserModel;
+  };
+}
+
+export default class PaymentHistorySectionComponent extends Component<Signature> {
+  @service declare store: Store;
+
+  rippleSpinnerImage = rippleSpinnerImage;
+  @tracked charges: ChargeModel[] = [];
+  @tracked errorMessage: string | null = null;
+  @tracked isLoading = true;
+
+  constructor(owner: unknown, args: Signature['Args']) {
+    super(owner, args);
+    this.loadCharges();
+  }
+
+  async loadCharges() {
+    this.isLoading = true;
+    this.errorMessage = null;
+
+    try {
+      const result = await this.store.query('charge', {
+        filter: { user_id: this.args.user.id },
+      });
+      this.charges = result.toArray();
+    } catch (error) {
+      console.error('Failed to fetch charges:', error);
```

Comment:
Can remove `console.error`, we'll never use this - `Sentry.captureException` is sufficient